lib/_http_server.js

```diff
@@ -904,6 +907,17 @@ function parserOnIncoming(server, socket, state, req, keepAlive) {
          resOnFinish.bind(undefined,
                           req, res, socket, state, server));
 
+  if (typeof server.maxRequestsPerSocket === 'number'
+    && (req.httpVersionMajor === 1 && req.httpVersionMinor === 1)) {
+
+    if (server.maxRequestsPerSocket < ++state.requestsCount) {
+      res.writeHead(503);
+      res.end();
```

Comment:
Is connection close correctly set here? I think shouldKeepAlive should be set to false.